DataProcessing/datax-host/src/main/scala/datax/telemetry/MetricLogger.scala

```diff
@@ -66,6 +69,8 @@ class MetricLogger(name: String, redis: RedisConnection, eventhub: EventHubSende
         HttpPoster.postEvents((fullJson).toSeq, httpEndpoint, Some(header),"metricLogger")
       }
 
+      AppInsightLogger.trackEvent(ProductConstant.ProductRoot + "/BatchMetrics", Map("timestamp" -> timestamp.toString), metrics.toMap)
```

Comment:
this is time taken to log a metric?  this may be optional

Reply:
If Instrumentation key is not provided, then trackEvent would do nothing (no exception is raised in such case)